internal/appsec/waf_test.go

```diff
@@ -46,6 +46,44 @@ func TestWAF(t *testing.T) {
 	srv := httptest.NewServer(mux)
 	defer srv.Close()
 
+	// This needs to be the first test run in order to retrieve WAF metrics correctly. All the tests here
+	//are using the same WAF handle, meaning that ruleset metrics will be sent only once on the first request.
+	t.Run("metrics", func(t *testing.T) {
```

Comment:
This makes me think you should rather do a simple unit test of `addRulesetInfoTags()` where you'd be able to do the same assertion about the `string` type if you want.

I'll work on a proper a system test to check the json value.